spec/models/notification_service/slack_service_spec.rb

```diff
@@ -110,4 +112,15 @@
       service.create_notification(problem)
     end
   end
+
+  context 'with only_production' do
+    let(:production_only) { true }
+    let(:notice) { Fabricate :notice, server_environment: { 'environment-name' => 'development' } }
+
+    it "should ignore a notification to Slack" do
+      expect(HTTParty).to_not receive(:post)
+
+      service.create_notification(problem)
+    end
```

Comment:
Another test of the opposite case would make a lot of sense here - one where ` 'environment-name' => 'production'`, to actually test the flow works as intended.